src/communication/tlschannel.cpp

```diff
@@ -50,6 +50,8 @@ Error TLSChannel::SetTLSConfig(const String& certType)
 {
     LOG_DBG() << "Set TLS config: name=" << mName << ", certType=" << certType;
 
+    Cleanup();
```

Comment:
As I remember there were some issues cleaning up not setup TLS config. So, cleanup only if mCertType not empty and clear it in cleanup for example.